Orm/Xtensive.Orm/Sql/Dml/Hints/SqlForceJoinOrderHint.cs

```diff
@@ -18,10 +18,9 @@ public class SqlForceJoinOrderHint : SqlHint
     /// </summary>
     public IEnumerable<SqlTable> Tables { get { return tables; } }
 
-    internal override object Clone(SqlNodeCloneContext context) =>
-      context.NodeMapping.TryGetValue(this, out var clone)
-        ? clone
-        : context.NodeMapping[this] = new SqlForceJoinOrderHint(tables?.Select(table => (SqlTable) table.Clone()).ToArray());
+    internal override SqlForceJoinOrderHint Clone(SqlNodeCloneContext context) =>
+      context.GetOrAdd(this, static (t, c) =>
+        new SqlForceJoinOrderHint(t.tables?.Select(table => (SqlTable) table.Clone()).ToArray()));
```

Comment:
Should we do following
```
    new SqlForceJoinOrderHint((t.tables is null)
          ? Array.Empty<SqlTable>()
          : t.tables.Select(table => (SqlTable) table.Clone()).ToArray(t.tables.Length)));
```

I didn't add it as suggestion deliberately because ```using Xtensive.Core;``` has to be added to namespaces

Reply:
> Should we do following:

No. if the original `table == nul` then clone's `table` should be null, not empty array.

Reply:
Yes, you're right, I should've replaced Array.Empty with null
```
new SqlForceJoinOrderHint((t.tables is null)
          ? null
          : t.tables.Select(table => (SqlTable) table.Clone()).ToArray(t.tables.Length)));
```
I wanted to use .ToArray(lenght) extension.

Reply:
Actually it is not necessary
`.ToArray()` extension smart enough to extract `.Count()` of its source.Select().

This is the effective implementation:
(https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Select.SpeedOpt.cs#L74)